src/ychaos/agents/coordinator.py

```diff
@@ -81,13 +81,13 @@ def callable_hook(agent_name: str): ...
     ---
     """
 
-    __hook_events__ = (
-        "on_attack_start",
-        "on_attack_completed",
-        "on_each_agent_start",
-        "on_each_agent_teardown",
-        "on_each_agent_stop",
-    )
+    __hook_events__ = {
+        "on_attack_start": EventHook.CallableType(),
+        "on_attack_completed": EventHook.CallableType(),
+        "on_each_agent_start": EventHook.CallableType(str),
+        "on_each_agent_teardown": EventHook.CallableType(str),
+        "on_each_agent_stop": EventHook.CallableType(str),
```

Comment:
Repetition can be reduced to just a tuple, instead of calling the same `classmethod`!. And one classmethod can take this tuple and construct a Type object
```suggestion
        "on_attack_start": (),
        "on_attack_completed": (),
        "on_each_agent_start": (str, ),
        "on_each_agent_teardown": (str, ),
        "on_each_agent_stop": (str, ),
```